### DISCUSSION TITLE
A weird preprocessing error

### DISCUSSION THREAD

**Discussion by meakbiyik** [stocnet/goldfish#49] [Q&A] [answered]
Hi all, I got a weird error during preprocessing phase and cannot seem to find the reason. Here's the code and backtrace:

```r
rate_model_preprocessing <- estimate(
    rate_formula,
    preprocessingOnly = TRUE,
    model = "DyNAM",
    subModel = "rate"
)
```
```
Error in object[[name, exact = TRUE]] : subscript out of bounds
Backtrace:
1: getElement(get(row$nodeset, envir = envir), row$attribute)
2: getElementFromDataObjectTable(objectNameTable, envir = envir)
3: imputeMissingData(objectsEffectsLink, envir = prepEnvir)
4: preprocess(model, subModel, events = events, effects = effects, 
5: estimate.formula(rate_formula, preprocessingOnly = TRUE, model = "DyNAM",
6: estimate(rate_formula, preprocessingOnly = TRUE, model = "DyNAM", 
```

I know that this is not much to work on, but I am unfortunately unable to share the data and for all purposes, it looks fine. At this point, I am just looking for an idea: this seems to be somewhat connected to #5 but my R knowledge is not good enough to understand what is going on.

Thanks in advance!

**Comment by meakbiyik** [comment, marked as answer]
Okay, this issue has resolved itself -apparently I failed to see a mistake at an unrelated location in the code. Though, it seems the variable name for the nodes is somehow important, as assigning it to a variable named "nodes" gave me quite a headache. Will try to see if I can reproduce that and raise an issue later.

**Reply by auzaheta** [reply]
Issue #54 relates with this problem.
